### PR TITLE
Update documentation for repeating events

### PR DESCRIPTION
**Pull request by ezraporter**
# Description
Initial PR for documentation updates related to repeating events. So far I've only made change to the glossary but plan on continuing to update this.

`devtools::build_site()` to see the changes

# Proposed Changes 
List changes below in bullet format:

- Update glossary to include information about repeating events

### PR Checklist
Before submitting this PR, please check and verify below that the submission meets the below criteria:

- [ ] New/revised functions have associated tests
- [ ] New/revised functions that update downstream outputs have associated static testing files (`.RDS`) updated under `inst/testdata/create_test_data.R`
- [ ] New tests that make API calls use `httptest::with_mock_api` and any new mocks were added to `tests/testthat/fixtures/create_httptest_mocks.R`
- [ ] New/revised functions use appropriate naming conventions
- [ ] New/revised functions don't repeat code
- [ ] Code changes are less than **250** lines total
- [ ] Issues linked to the PR using [GitHub's list of keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [ ] The appropriate reviewer is assigned to the PR
- [ ] The appropriate developers are assigned to the PR
- [ ] Pre-release package version incremented using `usethis::use_dev_version()`

# Code Review
This section to be used by the reviewer and developers during Code Review after PR submission

### Code Review Checklist

- [ ] I checked that new files follow naming conventions and are in the right place
- [ ] I checked that documentation is complete, clear, and without typos
- [ ] I added/edited comments to explain "why" not "how"
- [ ] I checked that all new variable and function names follow naming conventions
- [ ] I checked that new tests have been written for key business logic and/or bugs that this PR fixes
- [ ] I checked that new tests address important edge cases

### CODE REVIEW

File: vignettes/glossary.Rmd [hunk 1]
```diff
@@ -139,11 +143,11 @@ A function provided by REDCapTidieR designed to help turning [field labels](#fie
...
+The level of detail that a specific [row](#row) in a [data tibble](#data-tibble) represents. This depends on the [structure](#structure) of the [project](#project) ([classic](#classic-project) vs. [longitudinal](#longitudinal-project) vs longitudinal with [arms](#arm)) and the structure of the [instrument](#instrument) ([nonrepeating](#nonrepeating-instrument) vs [repeating](#repeating-instrument)). For example, a data tibble containing data from a nonrepeating instrument in a longitudinal project with two arms has a granularity of one row per [record](#record) per [event](#event) per arm. See also: the section [Longitudinal REDCap projects](https://chop-cgtinformatics.github.io/REDCapTidieR/articles/diving_deeper.html#longitudinal-redcap-projects) in the [Diving Deeper vignette](https://chop-cgtinformatics.github.io/REDCapTidieR/articles/diving_deeper.html). <a onclick="history.back()">↩︎</a>
```
Comment (ezraporter) [inline vignettes/glossary.Rmd:146]: Do we need to update this further to reflect the fact that granularity depend on the structure of the project, the structure of the instrument, **and the structure of the events the instrument is associated with**?

File: vignettes/glossary.Rmd [hunk 2]
```diff
@@ -139,11 +143,11 @@ A function provided by REDCapTidieR designed to help turning [field labels](#fie
...
+A [column](#column) in the [data tibble](#data-tibble) that serves to partially identify the entity described in a [row](#row). The [record](#record) ID column is always present in the data tibble. Depending on the [structure](#structure) of the [project](#project) ([classic](#classic-project) vs. [longitudinal](#longitudinal-project) vs. longitudinal with [arms](#arm)) and the structure of the [instrument](#instrument) ([nonrepeating](#nonrepeating-instrument) vs [repeating](#repeating-instrument)) there may be additional identifier columns, including `redcap_event`, `redcap_arm`, `redcap_form_instance`, and `redcap_event_instance`. Taken together, the identifier columns form a [composite primary key](#composite-primary-key). See also: the section [Longitudinal REDCap projects](https://chop-cgtinformatics.github.io/REDCapTidieR/articles/diving_deeper.html#longitudinal-redcap-projects) in the [Diving Deeper vignette](https://chop-cgtinformatics.github.io/REDCapTidieR/articles/diving_deeper.html). <a onclick="history.back()">↩︎</a>
```
Comment (ezraporter) [inline vignettes/glossary.Rmd:150]: Same as above here

**Comment by ezraporter** [comment]
@rsh52 @skadauke haven't done too much here but wondering what your thoughts are on splitting the "repeating" and "nonrepeating" sections of the glossary into "repeating event", "reapeating instrument", "nonrepeating event", and "nonreapeating instrument".

I think option 2 is to keep it with two sections and alter them to say something like:

"In the context of an instrument ... . In the context of an event ... ."

File: vignettes/glossary.Rmd (hunk 1 above)
Comment (rsh52) [reply vignettes/glossary.Rmd:146]: Yes I think so, it should be a pretty simple addition since we're referring to the "data tibble" specifically here. It does make me wonder if any considerations need to be made to the supertibble since we have a `structure` column only useful for/indicative of forms, but I think that's ok since the supertibble is displayed on the basis of forms.

**Review by rsh52 — COMMENTED** [review]
Good start, thanks for kicking this off. We also need to fix the diving deeper vignette table, might have gotten messed up during the find and replace I did:

image

**Review by rsh52 — APPROVED** [review]
